File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.js
function solution(ingredient) {
    const stack = Array(ingredient.length).fill(0);
    let top = -1;
    let cnt = 0;
    
    ingredient.forEach(v => {
        top++;
        stack[top] = v;
        
        while (top > 2) {
            if (stack[top - 3] === 1 &&
               stack[top - 2] === 2 &&
               stack[top - 1] === 3 &&
               stack[top] === 1) {
                cnt++;
                top -= 4;
            } else {
                break;
            }
        }
    })
    
    return cnt;
}

File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.py
def solution(ingredient):
    stack = [0] * len(ingredient)
    top = -1
    cnt = 0

    for v in ingredient:
        top += 1
        stack[top] = v

        while top > 2:
            if (stack[top - 3] == 1 and
                stack[top - 2] == 2 and
                stack[top - 1] == 3 and
                stack[top] == 1):
                cnt += 1
                top -= 4
            else:
                break

    return cnt
